### ISSUE TITLE
Return values that are by-ref are not displayed correctly

### ISSUE THREAD

**Issue by pardeike** [dotnet/docfx#4284]
**Operation System**: `Windows` (Managed Azure Host)

**DocFX Version Used**: latest release (see log below)

**Template used**: `default`

**Steps to Reproduce**:

C# source code:
<https://github.com/pardeike/Harmony/blob/57f29f512f959985a899094a2dbbe0a19e12793a/Harmony/Tools/AccessTools.cs#L784>

Generated API:
<https://harmony.pardeike.net/api/Harmony.AccessTools.html#Harmony_AccessTools_FieldRefAccess__2___0_System_String_>

Configuration:
<https://github.com/pardeike/Harmony/blob/master/Harmony/Documentation/docfx.json>

**Expected Behavior**:

```
public static ref U FieldRefAccess<T, U>(T instance, string fieldName)
```

**Actual Behavior**:

```
public static U FieldRefAccess<T, U>(T instance, string fieldName)
```

Azure DevOps build log:
```
2019-03-26T22:31:34.0099677Z ##[section]Starting: Create DocFx Documentation
2019-03-26T22:31:34.0200102Z ==============================================================================
2019-03-26T22:31:34.0201340Z Task         : Create DocFx Documentation
2019-03-26T22:31:34.0201640Z Description  : Runs the DocFx tools to generate static site documentation
2019-03-26T22:31:34.0202056Z Version      : 0.1.19
2019-03-26T22:31:34.0202468Z Author       : 
2019-03-26T22:31:34.0202841Z Help         : 
2019-03-26T22:31:34.0203196Z ==============================================================================
2019-03-26T22:31:34.3004746Z D:\_work\_tasks\DocFxTask_e0923c33-09f9-43a0-a8af-86b7d0d76b03\0.1.19\tools D:\_work\1\s
2019-03-26T22:31:34.3056060Z [command]D:\_work\_tasks\DocFxTask_e0923c33-09f9-43a0-a8af-86b7d0d76b03\0.1.19\tools\nuget.exe install docfx.console
2019-03-26T22:31:52.1600106Z Feeds used:
2019-03-26T22:31:52.1602074Z   https://api.nuget.org/v3/index.json
2019-03-26T22:31:52.1603053Z 
2019-03-26T22:31:52.1603936Z   GET https://api.nuget.org/v3/registration3-gz/docfx.console/index.json
2019-03-26T22:31:52.1605103Z   OK https://api.nuget.org/v3/registration3-gz/docfx.console/index.json 117ms
2019-03-26T22:31:52.1605813Z 
2019-03-26T22:31:52.1606779Z 
2019-03-26T22:31:52.1607565Z Attempting to gather dependency information for package 'docfx.console.2.40.12' with respect to project 'D:\_work\_tasks\DocFxTask_e0923c33-09f9-43a0-a8af-86b7d0d76b03\0.1.19\tools', targeting 'Any,Version=v0.0'
2019-03-26T22:31:52.1608639Z Gathering dependency information took 145.62 ms
2019-03-26T22:31:52.1609380Z Attempting to resolve dependencies for package 'docfx.console.2.40.12' with DependencyBehavior 'Lowest'
2019-03-26T22:31:52.1610253Z Resolving dependency information took 0 ms
2019-03-26T22:31:52.1611693Z Resolving actions to install package 'docfx.console.2.40.12'
2019-03-26T22:31:52.1612771Z Resolved actions to install package 'docfx.console.2.40.12'
2019-03-26T22:31:52.1613618Z Retrieving package 'docfx.console 2.40.12' from 'nuget.org'.
2019-03-26T22:31:52.1614574Z   GET https://api.nuget.org/v3-flatcontainer/docfx.console/2.40.12/docfx.console.2.40.12.nupkg
2019-03-26T22:31:52.1615492Z   OK https://api.nuget.org/v3-flatcontainer/docfx.console/2.40.12/docfx.console.2.40.12.nupkg 398ms
2019-03-26T22:31:52.1616211Z Installing docfx.console 2.40.12.
2019-03-26T22:31:52.1616981Z Adding package 'docfx.console.2.40.12' to folder 'D:\_work\_tasks\DocFxTask_e0923c33-09f9-43a0-a8af-86b7d0d76b03\0.1.19\tools'
2019-03-26T22:31:52.1617766Z Added package 'docfx.console.2.40.12' to folder 'D:\_work\_tasks\DocFxTask_e0923c33-09f9-43a0-a8af-86b7d0d76b03\0.1.19\tools'
2019-03-26T22:31:52.1618552Z Successfully installed 'docfx.console 2.40.12' to D:\_work\_tasks\DocFxTask_e0923c33-09f9-43a0-a8af-86b7d0d76b03\0.1.19\tools
2019-03-26T22:31:52.1619288Z Executing nuget actions took 4.55 sec
2019-03-26T22:31:52.1621084Z D:\_work\1\s
2019-03-26T22:31:52.1663420Z [command]D:\_work\_tasks\DocFxTask_e0923c33-09f9-43a0-a8af-86b7d0d76b03\0.1.19\tools\docfx.console.2.40.12\tools\docfx.exe D:\_work\1\s\Harmony\Documentation\docfx.json
2019-03-26T22:32:19.2752087Z [19-03-26 10:31:55.477]Info:[MetadataCommand.ExtractMetadata]Using dotnet C:\Program Files\dotnet\sdk\2.2.104\MSBuild.dll as inner compiler.
2019-03-26T22:32:19.2753596Z [19-03-26 10:31:56.829]Info:[MetadataCommand.ExtractMetadata]Loading projects...
2019-03-26T22:32:19.2754751Z [19-03-26 10:32:00.903]Warning:[MetadataCommand.ExtractMetadata](D:/_work/1/s/Harmony/Harmony.csproj)Workspace failed with: [Failure] Msbuild failed when processing the file 'D:\_work\1\s\Harmony\Harmony.csproj' with message: C:\Program Files\dotnet\sdk\2.2.104\Microsoft.Common.CurrentVersion.targets: (1179, 5): .NET Framework v3.5 Service Pack 1 was not found. In order to target ".NETFramework,Version=v3.5", .NET Framework v3.5 Service Pack 1 or later must be installed.
2019-03-26T22:32:19.2756175Z C:\Program Files\dotnet\sdk\2.2.104\Microsoft.Common.CurrentVersion.targets: (1179, 5): The reference assemblies for framework ".NETFramework,Version=v3.5" were not found. To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed. Note that assemblies will be resolved from the Global Assembly Cache (GAC) and will be used in place of reference assemblies. Therefore your assembly may not be correctly targeted for the framework you intend.
2019-03-26T22:32:19.2758970Z [19-03-26 10:32:01.388]Info:[MetadataCommand.ExtractMetadata]Generating metadata for each project...
2019-03-26T22:32:19.2759753Z [19-03-26 10:32:09.466]Info:[MetadataCommand]Completed Scope:MetadataCommand in 14587.0628 milliseconds.
2019-03-26T22:32:19.2760994Z [19-03-26 10:32:09.686]Info:[BuildCommand]6 plug-in(s) loaded.
2019-03-26T22:32:19.2761800Z [19-03-26 10:32:09.764]Info:[BuildCommand]Markdown engine is markdig
2019-03-26T22:32:19.2762712Z [19-03-26 10:32:10.217]Info:[BuildCommand]Cannot load build info: 'build.info' not found under 'D:\_work\1\s\Harmony\Documentation\obj\.cache\build'
2019-03-26T22:32:19.2763544Z [19-03-26 10:32:10.264]Info:[BuildCommand.BuildCore.Build Document]Max parallelism is 2.
2019-03-26T22:32:19.2764423Z [19-03-26 10:32:10.389]Info:[BuildCommand.BuildCore.Build Document.Prepare.CreateIncrementalBuildContext]Group  will be built fully.
2019-03-26T22:32:19.2765148Z [19-03-26 10:32:10.389]Info:[BuildCommand.BuildCore.Build Document.Prepare.CreateIncrementalBuildContext]The reason of full building under group  is: Cannot build incrementally because last build info is missing.
2019-03-26T22:32:19.2765978Z [19-03-26 10:32:11.438]Info:[BuildCommand.BuildCore.Build Document.CompilePhaseHandlerWithIncremental.ManagedReferenceDocumentProcessor]Building 48 file(s) in ManagedReferenceDocumentProcessor(BuildManagedReferenceDocument=>ValidateManagedReferenceDocumentMetadata=>ApplyOverwriteDocumentForMref=>FillReferenceInformation)...
2019-03-26T22:32:19.2766821Z [19-03-26 10:32:11.438]Info:[BuildCommand.BuildCore.Build Document.CompilePhaseHandlerWithIncremental.ResourceDocumentProcessor]Building 4 file(s) in ResourceDocumentProcessor(ValidateResourceMetadata)...
2019-03-26T22:32:19.2767631Z [19-03-26 10:32:11.438]Info:[BuildCommand.BuildCore.Build Document.CompilePhaseHandlerWithIncremental.ConceptualDocumentProcessor]Building 14 file(s) in ConceptualDocumentProcessor(BuildConceptualDocument=>CountWord=>ValidateConceptualDocumentMetadata)...
2019-03-26T22:32:19.2768481Z [19-03-26 10:32:11.438]Info:[BuildCommand.BuildCore.Build Document.CompilePhaseHandlerWithIncremental.TocDocumentProcessor]Building 3 file(s) in TocDocumentProcessor(BuildTocDocument)...
2019-03-26T22:32:19.2769270Z [19-03-26 10:32:15.685]Warning:[BuildCommand.BuildCore.Build Document.LinkPhaseHandlerWithIncremental.ConceptualDocumentProcessor.Save](index.md#L22)Invalid file link:(~/articles/intro.html).
2019-03-26T22:32:19.2770036Z [19-03-26 10:32:15.685]Warning:[BuildCommand.BuildCore.Build Document.LinkPhaseHandlerWithIncremental.ConceptualDocumentProcessor.Save](index.md#L22)Invalid file link:(~/api/index.html).
2019-03-26T22:32:19.2771270Z [19-03-26 10:32:15.700]Info:[BuildCommand.BuildCore.Build Document.LinkPhaseHandlerWithIncremental.UpdateContext]0 external references found in 1 xref maps.
2019-03-26T22:32:19.2772054Z [19-03-26 10:32:16.591]Info:[BuildCommand.BuildCore.Build Document.LinkPhaseHandlerWithIncremental.Apply Templates]Applying templates to 69 model(s)...
2019-03-26T22:32:19.2773107Z [19-03-26 10:32:18.610]Info:[BuildCommand.BuildCore.Build Document]XRef map exported.
2019-03-26T22:32:19.2773841Z [19-03-26 10:32:19.112]Info:[BuildCommand.Postprocess]Manifest file saved to manifest.json.
2019-03-26T22:32:19.2774897Z [19-03-26 10:32:19.205]Info:[BuildCommand]Completed building documents in 9510.9973 milliseconds.
2019-03-26T22:32:19.2775450Z [19-03-26 10:32:19.205]Info:[BuildCommand]Completed Scope:BuildCommand in 9749.1289 milliseconds.
2019-03-26T22:32:19.2776018Z [19-03-26 10:32:19.205]Info:Completed in 24342.2903 milliseconds
2019-03-26T22:32:19.2777168Z 
2019-03-26T22:32:19.2777640Z 
2019-03-26T22:32:19.2778132Z Build succeeded with warning.
2019-03-26T22:32:19.2778825Z [19-03-26 10:32:19.221]Warning:[MetadataCommand.ExtractMetadata](D:/_work/1/s/Harmony/Harmony.csproj)Workspace failed with: [Failure] Msbuild failed when processing the file 'D:\_work\1\s\Harmony\Harmony.csproj' with message: C:\Program Files\dotnet\sdk\2.2.104\Microsoft.Common.CurrentVersion.targets: (1179, 5): .NET Framework v3.5 Service Pack 1 was not found. In order to target ".NETFramework,Version=v3.5", .NET Framework v3.5 Service Pack 1 or later must be installed.
2019-03-26T22:32:19.2779810Z C:\Program Files\dotnet\sdk\2.2.104\Microsoft.Common.CurrentVersion.targets: (1179, 5): The reference assemblies for framework ".NETFramework,Version=v3.5" were not found. To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed. Note that assemblies will be resolved from the Global Assembly Cache (GAC) and will be used in place of reference assemblies. Therefore your assembly may not be correctly targeted for the framework you intend.
2019-03-26T22:32:19.2781255Z [19-03-26 10:32:19.221]Warning:[BuildCommand.BuildCore.Build Document.LinkPhaseHandlerWithIncremental.ConceptualDocumentProcessor.Save](index.md#L22)Invalid file link:(~/articles/intro.html).
2019-03-26T22:32:19.2782083Z [19-03-26 10:32:19.221]Warning:[BuildCommand.BuildCore.Build Document.LinkPhaseHandlerWithIncremental.ConceptualDocumentProcessor.Save](index.md#L22)Invalid file link:(~/api/index.html).
2019-03-26T22:32:19.2782769Z 	3 Warning(s)
2019-03-26T22:32:19.2783381Z 	0 Error(s)
2019-03-26T22:32:19.2847556Z ##[section]Finishing: Create DocFx Documentation
```

**Comment by carlreinke** [comment]
`ref struct`s are also not presented correctly.

**Comment by jo3w4rd** [comment]
We are seeing this too -- any update? ref return statements are becoming more common (at least in our code base).

**Comment by yufeih** [comment]
Fixed by #6805
